vcenter_operator/templates.py

```diff
@@ -223,17 +218,149 @@ def _create_custom_resource_definitions(self):
 
         api = client.ApiextensionsV1Api()
         self._crd = \
-            CustomResourceDefinitionLoader._custom_resource_definition()
+            TemplateCRDLoader._custom_resource_definition()
+
         try:
             api.create_custom_resource_definition(self._crd)
         except client.rest.ApiException:
-            pass
+            LOG.exception("Failed to create custom resource definition vcenter-template")
+
+
+class ServiceUserCRDLoader(PollingLoader):
+    API_GROUP = "vcenter-operator.stable.sap.cc"
+
+    def __init__(self):
+        self.mapping = {}
+        self._crd = None
+
+    def get_source(self, environment, template):
+        if template in self.mapping:
+            version, source, jinja2_options, owner = self.mapping[template]
+            restore_defaults(environment)
+
+            for k in jinja2_options:
+                if hasattr(environment, k):
+                    store_default(environment, k)
+                    setattr(environment, k, jinja2_options[k])
+
+            return (
+                source,
+                None,
+                lambda: template in self.mapping
+                and (version, source, jinja2_options, owner) == self.mapping.get(template),
+            )
+        raise TemplateNotFound(template)
+
+    def list_templates(self):
+        return sorted(self.mapping)
+
+    def get_source_owner(self, template_name):
+        if template_name not in self.mapping:
+            return None
+
+        _, _, _, owner = self.mapping[template_name]
+        return owner
+
+    def _read_options_v1(self, item):
+        options = {"scope": item["metadata"]["scope"], "jinja2_options": item["metadata"].get("jinja2_options", {})}
+        return options
+
+    def _read_options_v2(self, item):
+        return item["options"]
 
+    def poll(self):
+        if not self._crd:
+            self._create_custom_resource_definitions()
+
+        api = client.CustomObjectsApi()
+
+        mapping = dict()
+
+        group = self._crd.spec["group"]
+        plural = self._crd.spec["names"]["plural"]
+        version = self._crd.spec["version"]
+        try:
+            resp = api.list_cluster_custom_object(group, version, plural)
+        except (client.rest.ApiException, urllib3.exceptions.MaxRetryError) as e:
+            raise CustomResourceDefinitionLoadingError(e)
+
+        for item in resp["items"]:
+            # Get service-user template names
+            try:
+                metadata = item["metadata"]
+                version = metadata["resourceVersion"]
+                name = metadata["name"]
+                namespace = metadata["namespace"]
+                service_username_template = item["spec"]["template"]["username"]
```

Comment:
Can we build "validation" for "username" only existing in one CR?

Alternatively: Take the "first one" from stable ordering and raise an error for the other one.